Storage/transaction_policy.h

```diff
@@ -146,19 +149,35 @@ struct Synchronous final {
       return future;                                                                         // LCOV_EXCL_LINE
     }
     bool successful = false;
+    T_RESULT f1_result;
     try {
-      f2(f1());
+      f1_result = f1();
```

Comment:
Agree. However, in `Transaction(F&& f)` above we have to declare it outside, because we want to pass it as the result. So I decided to make it same here.